src/main/java/jenkins/plugins/publish_over_ssh/BapSshCredentials.java

```diff
@@ -39,10 +39,17 @@
 
     private final String username;
 
+    private final boolean injectCredentials;
+
     @DataBoundConstructor
-    public BapSshCredentials(final String username, final String encryptedPassphrase, final String key, final String keyPath) {
+    public BapSshCredentials(final String username, final String encryptedPassphrase, final String key, final String keyPath, boolean injectCredentials) {
```

Comment:
Normally you should create a new constructor and mark the old one as deprecated. Someone might be using this one.